README.md

```diff
@@ -496,63 +491,86 @@ The `auth` field MUST be contain an [Authorization] which signs over the `&Task`
 
 Concretely, this means that the `&Task` MUST be present in the associated `auth`'s `scope` field. An `Invocation` where the associated [Authorization] does not include the [Task] in the `scope` MUST be considered invalid.
 
-## 5.1 Schema
+## 5.1 Context
 
 ```ipldsch
-type Invocation struct {
-  v       SemVer
-
+type Context struct {
   run     &Task
+  meta    {String : Any}
+  prf     [&UCAN]
+
   # Receipt of the invocation that caused this invocation
   cause   optional &Invocation
+}
+```
 
-  # Task authorization.
-  auth    &Authorization
+### 5.1.1 Task
 
-  meta    {String : any}
+The `run` field MUST contain a link to the [Task] to be run.
 
-  prf     [&UCAN]
-}
+### 5.1.2 Metadata
+
+The OPTIONAL `meta` field MAY be used to include human-readable descriptions, tags, execution hints, resource limits, and so on. If present, the `meta` field MUST contain a map with string keys. The contents of the map are left undefined to encourage extensible use.
+
+If `meta` field is not present, it is implicitly a `unit` represented as an empty map.
 
-type SemVer string
+### 5.1.3 Proofs
+
+The `prf` field MUST contain links to any UCANs that provide the authority to perform this task. All of the outermost proofs MUST have `aud` field set to the [Executor]'s DID. All of the outmost proofs MUST have `iss` field set to the [Invoker]'s DID.
+
+### 5.1.4 Cause
+
+[Task]s MAY be invoked as an effect caused by a prior [Invocation]. Such [Invocation]s SHOULD have a `cause` field set to the [Receipt] link of the [Invocation] that caused it. The linked [Receipt] MUST have an `Effect` (the `fx` field) containing invoked [Task] in the `run` field.
+
+## 5.2 (Signed) Invocation
+
+An `Invocation` is a signed `Context`.
+
+```ipldsch
+type Invocation struct {
+  ctx     Context
```

Comment:
Reading this it does feel like lifting run field would be clearer as in signed request to run task with a given context

Reply:
Irakli and I talked through this on a call on Friday. The counterargument to pulling the `run` field up one level is that IPVM wants to sign over both the raw function and its contextual info (e.g. gas limit) beyond the sementic hash (which we want to key the function by).

If we pulled `run` up to this level:

```ocaml
type Invocation struct {
  ctx Context
  sig Signature
}
```